handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -215,6 +215,43 @@ int calculateOutNetBufSize(SslHandler handler, int pendingBytes) {
                 return ReferenceCountedOpenSslEngine.calculateOutNetBufSize(pendingBytes);
             }
         },
+        CONSCRYPT(true, COMPOSITE_CUMULATOR) {
+            @Override
+            SSLEngineResult unwrap(SslHandler handler, ByteBuf in, int readerIndex, int len, ByteBuf out)
+                    throws SSLException {
+                int nioBufferCount = in.nioBufferCount();
+                int writerIndex = out.writerIndex();
+                final SSLEngineResult result;
+                if (nioBufferCount > 1) {
+                    /**
+                     * If {@link org.conscrypt.OpenSslEngine} is in use,
+                     * we can use a special
+                     * {@link org.conscrypt.OpenSslEngine#unwrap(ByteBuffer[], ByteBuffer[])} method
+                     * that accepts multiple {@link ByteBuffer}s without additional memory copies.
+                     */
+                    org.conscrypt.OpenSSLEngineImpl conscryptEngine =
+                            ((ConscryptAlpnSslEngine) handler.engine).getWrappedConscryptEngine();
+                    try {
+                        handler.singleBuffer[0] = toByteBuffer(out, writerIndex, out.writableBytes());
+                        result = conscryptEngine.unwrap(in.nioBuffers(readerIndex, len), handler.singleBuffer);
+                    } finally {
+                        handler.singleBuffer[0] = null;
+                    }
+                } else {
+                    result = handler.engine.unwrap(toByteBuffer(in, readerIndex, len),
+                            toByteBuffer(out, writerIndex, out.writableBytes()));
+                }
+                out.writerIndex(writerIndex + result.bytesProduced());
+                return result;
+            }
+
+            @Override
+            int calculateOutNetBufSize(SslHandler handler, int pendingBytes) {
```

Comment:
@nmittler - note that I found a bug related to this method ... see https://github.com/netty/netty/commit/53fc69390130f7a5a33d454a59ad214f6b4176f5